stan/math/prim/fun/isinf.hpp

```diff
@@ -17,10 +19,9 @@ namespace math {
  * @param[in] v argument
  * @return true if argument is infinite
  */
-template <typename ADType, require_autodiff_t<ADType>...>
-inline bool isinf(ADType&& v) {
-  using std::isinf;
-  return isinf(v.val());
+template <typename T>
+inline bool isinf(const T& v) {
+  return is_inf(v);
```

Comment:
Why the change to is_inf?

Reply:
Because (a) I couldn't resolve all the dependencies otherwise, and (b) we don't need two definitions.  So I killed two birds with one stone and delegated.

Reply:
And if you can figure out how to put the constraints back and still satisfy all the header and unit tests, then that'd also be OK, though I'd really just as soon not rely on something like this.

Reply:
Looking through the specializations of is_inf I think this is fine